File: game-server/src/main/java/org/forfun/mmorpg/game/scene/factory/MonsterCreator.java
package org.forfun.mmorpg.game.scene.factory;

import org.forfun.mmorpg.game.database.config.domain.ConfigMonster;
import org.forfun.mmorpg.game.scene.actor.Monster;
import org.forfun.mmorpg.game.scene.actor.SceneActor;

public class MonsterCreator implements ActorCreator {

	@Override
	public SceneActor create(ActorPrototype prototype, Object... params) {
		ConfigMonster configMonster = (ConfigMonster)prototype;
		Monster monster = new Monster();
		monster.setId(configMonster.getId());
		monster.setName(configMonster.getName());
		
		return monster;
	}

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/scene/factory/MonsterCreator.kt
package org.forfun.mmorpg.game.scene.factory

import org.forfun.mmorpg.game.database.config.domain.ConfigMonster
import org.forfun.mmorpg.game.scene.actor.Monster
import org.forfun.mmorpg.game.scene.actor.SceneActor

class MonsterCreator : ActorCreator {

    override fun create(prototype: ActorPrototype, vararg params: Any): SceneActor {
        val configMonster = prototype as ConfigMonster
        val monster = Monster()
        monster.modelId = configMonster.id
        monster.name = configMonster.name

        return monster
    }
}
